app/src/main/java/com/internship/move/feature/splash/SplashFragment.kt

```diff
@@ -15,15 +16,23 @@ class SplashFragment : Fragment(R.layout.fragment_splash) {
         super.onViewCreated(view, savedInstanceState)
 
         Handler(Looper.getMainLooper()).postDelayed({
-            findNavController().navigate(SplashFragmentDirections.actionSplashFragmentToOnboardingFragment())
+            navigateToNextFragment()
         }, SPLASH_NAV_DELAY)
 
-        if (lifecycle.currentState == Lifecycle.State.RESUMED)
+        if (lifecycle.currentState == Lifecycle.State.STARTED)
             Handler(Looper.getMainLooper()).postDelayed({
-                findNavController().navigate(SplashFragmentDirections.actionSplashFragmentToOnboardingFragment())
+                navigateToNextFragment()
             }, SPLASH_NAV_DELAY)
     }
 
+    private fun navigateToNextFragment() {
+        val pref = activity?.getPreferences(Context.MODE_PRIVATE)
+        if (pref?.getBoolean("IS_LOGGED", false) == true)
```

Comment:
You can simplify this by inlining it in the `if` call
`requireActivity().getPreferences(Context.MODE_PRIVATE).getBoolean(KEY_IS_USER_LOGGED_IN, false)`

Reply:
Also use `{ }` for if/else bodies